ark/spLDA/processing.py

```diff
@@ -1,86 +1,186 @@
 
     Returns:
-        pd.DataFrame:
-
-        - A formatted cell table for use in spatial-LDA analysis
+        dict:
 
+        - A dictionary of formatted cell tables for use in spatial-LDA analysis.  Each element
+        in the dictionary is a Dataframe corresponding to a single field of view.
     """
 
     # Check function arguments
-    check_format_cell_table_args(cell_table=cell_table, markers=markers, clusters=clusters, fovs=fovs)
+    check_format_cell_table_args(cell_table=cell_table, markers=markers, clusters=clusters)
 
     # Only keep columns relevant for spatial-LDA
-    all_columns = cell_table.columns
-    keep_columns = [
-        "point",
-        "label",
-        "cell_size",
-        "centroid-0",
-        "centroid-1",
-        "pixelfreq_hclust_cap",
-        "name"
-    ]
     if markers is not None:
-        keep_columns.append(markers)
-    drop_columns = [c for c in all_columns if c not in keep_columns]
-    cell_table = cell_table.drop(columns = drop_columns)
+        BASE_COLS.append(markers)
+    drop_columns = [c for c in cell_table.columns if c not in BASE_COLS]
+    cell_table = cell_table.drop(columns=drop_columns)
 
     # Rename columns
     cell_table = cell_table.rename(
         columns={
-            "cell_size": "area",
             "centroid-0": "x",
             "centroid-1": "y",
-            "pixelfreq_hclust_cap": "cluster_id",
-            "name": "cluster"
+            "FlowSOM_ID": "cluster_id",
+            "cluster_labels": "cluster"
         })
 
     # Create dictionary of FOVs
-    if fovs == "all_fovs":
-        fovs = [x + 1 for x in range(max(cell_table["point"]))]
+    fovs = np.unique(cell_table["SampleID"])
 
     fov_dict = {}
     for i in fovs:
-        df = cell_table[cell_table["point"] == i].drop(columns=["point", "label"])
+        df = cell_table[cell_table["SampleID"] == i].drop(
+            columns=["SampleID", "label"])
         if clusters is not None:
             df = df[df["cluster_id"].isin(clusters)]
-        df['is_index'] = True
-        df['isimmune'] = True
+        df["is_index"] = True
+        df["is_immune"] = True  # might remove this
         fov_dict[i] = df.reset_index(drop=True)
 
-    return(fov_dict)
+    # Save Arguments
+    fov_dict["fovs"] = fovs
+    fov_dict["markers"] = markers
+    fov_dict["clusters"] = clusters
 
+    return fov_dict
 
 
+def featurize_cell_table(cell_table, featurization=None, radius=100, cell_index=None,
+                         n_processes=None, train_frac=0.75):
+    """Calculates statistics for local cellular neighborhoods based on the specified features
+    and radius.
 
+    Args:
+        cell_table (dict):
+            A formatted cell table for use in spatial-LDA analysis. Specifically, this is the
+            output from :func:`~ark.spLDA.processing.format_cell_table`.
+        featurization (str):
+            One of four choices of featurization method, defaults to "cluster" if None is provided:
```

Comment:
Here's an example where setting the default to None isn't as useful. Because we know exactly what we're going to substitute None for ("cluster"), having None as a placeholder isn't better than just having the default be "cluster". When the default is None for all FOVs, it's a bit different, because the user isn't actually passing in all of the FOVs, they're passing in something which is telling us to use all the FOVs.